portability-worker/src/main/java/org/dataportabilityproject/worker/PortabilityInMemoryTransferCopier.java

```diff
@@ -0,0 +1,103 @@
+/*
+* Copyright 2018 The Data-Portability Project Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* https://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.dataportabilityproject.worker;
+
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.dataportabilityproject.spi.transfer.InMemoryTransferCopier;
+import org.dataportabilityproject.spi.transfer.provider.ExportResult;
+import org.dataportabilityproject.spi.transfer.provider.Exporter;
+import org.dataportabilityproject.spi.transfer.provider.Importer;
+import org.dataportabilityproject.spi.transfer.provider.TransferServiceProviderRegistry;
+import org.dataportabilityproject.spi.transfer.types.ContinuationData;
+import org.dataportabilityproject.spi.transfer.types.ExportInformation;
+import org.dataportabilityproject.types.transfer.PortableType;
+import org.dataportabilityproject.types.transfer.auth.AuthData;
+import org.dataportabilityproject.types.transfer.models.ContainerResource;
+import org.dataportabilityproject.types.transfer.models.DataModel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implementation of InMemoryTransferCopier.
+ */
+public class PortabilityInMemoryTransferCopier implements InMemoryTransferCopier {
+
+  private static final AtomicInteger COPY_ITERATION_COUNTER = new AtomicInteger();
+  private static final Logger logger = LoggerFactory.getLogger(PortabilityInMemoryTransferCopier.class);
+
+  @Override
+  public void copyDataType(TransferServiceProviderRegistry registry,
+      PortableType dataType,
+      String exportService,
+      AuthData exportAuthData,
+      String importService,
+      AuthData importAuthData,
+      String jobId) throws IOException {
+
+    Exporter<AuthData, DataModel> exporter = registry.getExporter(exportService, dataType);
+    Importer<AuthData, DataModel> importer = registry.getImporter(importService, dataType);
+
+    ExportInformation emptyExportInfo = new ExportInformation(null, null);
+    logger.debug("Starting copy job, id: {}, source: {}, destination: {}", jobId, exportService,
+        importService);
+    copy(exporter, importer, exportAuthData, importAuthData, emptyExportInfo);
+
+  }
+
+  private void copy(
```

Comment:
This method transfers data. It carries out an export from the given {@code exporter}, optionally starting at the point specified in the provided {@code exportInformation} then calls the {@code importer} with the exported data.  If more data is required to be exported, this method is called recursively with specific {@link ExportInformation} to continue the export process.

Reply:
Done.